evennia/contrib/rpg/rpsystem/rpsystem.py

```diff
@@ -1333,30 +1336,30 @@ def get_search_result(
         """
         # we also want to use the default search method
         search_obj = super().get_search_result
-        is_builder = self.locks.check_lockstring(self, "perm(Builder)")
+        is_builder = self.permissions.check("Builder")
+        results = []
 
-        if candidates:
-            candidates = parse_sdescs_and_recogs(
+        if candidates is not None:
```

Comment:
explicitly checks candidates being unset (`None`) versus just being an empty list - e.g. when searching for an object in an empty inventory inventory - avoiding *one* case where it was falling back to a global search when it shouldn't